### PR TITLE
ONYX 8733: Add tests for validate and decode jwt token

### PR DESCRIPTION
**Pull request by tzheleznyak**
### What does this PR do?
Add tests for authn jwt validate and decode

### What ticket does this PR close?
image

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb [hunk 1]
```diff
@@ -44,6 +62,10 @@ def rsa_key
     @rsa_key ||= OpenSSL::PKey::RSA.new(BITS_2048)
   end
 
+  def non_memoized_rsa_key
```
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb:65]: won't fix

[the remark this answers is not in the document]
Comment (tzheleznyak) [reply cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb:65]: won't fix

File: cucumber/authenticators_jwt/features/authn_jwt_validate_and_decode.feature [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I am the super-user
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    And I initialize JWKS endpoint with file "myJWKs.json"
+
+  Scenario: Signature error, kid not found
+    Given I issue unknoown kid JWT token:
```
Comment (Avi-Eyal) [inline cucumber/authenticators_jwt/features/authn_jwt_validate_and_decode.feature:40]: unknown*

File: cucumber/authenticators_jwt/features/step_definitions/jwt_jwks_steps.rb [hunk 3]
```diff
@@ -11,6 +11,24 @@
   )
 end
 
+Given(/I issue unknoown kid JWT token:/) do |token_body_string|
```
Comment (Avi-Eyal) [inline cucumber/authenticators_jwt/features/step_definitions/jwt_jwks_steps.rb:14]: *unknown

**Review by Avi-Eyal — DISMISSED** [review]
LGTM

**Review by Avi-Eyal — APPROVED** [review]
LGTM

**Pull request merged**